exporter/prometheusremotewriteexporter/helper.go

```diff
@@ -215,21 +220,35 @@ func getPromMetricName(metric *otlp.Metric, ns string) string {
 	return sanitize(b.String())
 }
 
-// Simple helper function that takes the <Signature String - *TimeSeries> map
-// and creates a WriteRequest from the struct -- can move to the helper.go file
-func wrapTimeSeries(tsMap map[string]*prompb.TimeSeries) (*prompb.WriteRequest, error) {
+// Helper function that takes the <Signature String - *TimeSeries> map
+// and creates a batch of WriteRequest from the struct
+func wrapAndBatchTimeSeries(tsMap map[string]*prompb.TimeSeries) ([]*prompb.WriteRequest, error) {
 	if len(tsMap) == 0 {
 		return nil, errors.New("invalid tsMap: cannot be empty map")
 	}
+
+	var requests []*prompb.WriteRequest
 	var tsArray []prompb.TimeSeries
+
 	for _, v := range tsMap {
 		tsArray = append(tsArray, *v)
+
+		if len(tsArray)%metricIntervalToCheckByteSize == 0 {
+			wrapped := convertTimeseriesToRequest(tsArray)
+
+			if b := proto.Size(wrapped); b >= maxBatchByteSize || len(tsArray) >= maxBatchMetric {
```

Comment:
Shouldn't we do the split before the size is not yet the maxBatchByteSize or length is bigger than maxBatchMetric?

In this case, we may still have individual write requests with larger payloads than the max limit.